core/src/main/java/brooklyn/entity/rebind/ActivePartialRebindIteration.java

```diff
@@ -0,0 +1,158 @@
+ */
+public class ActivePartialRebindIteration extends RebindIteration {
+
+    private static final Logger LOG = LoggerFactory.getLogger(ActivePartialRebindIteration.class);
+    
+    protected Iterator<BrooklynObject> objectsToRebindInitial;
+    protected Collection<BrooklynObject> objectsToRebindFinal;
+    protected List<CompoundTransformer> transformers = MutableList.of();
+    
+    public ActivePartialRebindIteration(RebindManagerImpl rebindManager, 
+            ManagementNodeState mode,
+            ClassLoader classLoader, RebindExceptionHandler exceptionHandler,
+            Semaphore rebindActive, AtomicInteger readOnlyRebindCount, PersistenceActivityMetrics rebindMetrics, BrooklynMementoPersister persistenceStoreAccess
+            ) {
+        super(rebindManager, mode, classLoader, exceptionHandler, rebindActive, readOnlyRebindCount, rebindMetrics, persistenceStoreAccess);
+    }
+
+    @Override
+    protected boolean isRebindingActiveAgain() {
+        return true;
+    }
+    
+    public void setObjectIterator(Iterator<BrooklynObject> objectsToRebind) {
+        this.objectsToRebindInitial = objectsToRebind;
+    }
+    
+    public void applyTransformer(CompoundTransformer transformer) {
+        if (transformer!=null)
+            transformers.add(transformer);
+    }
+    
+    @Override
+    protected void doRun() throws Exception {
+        checkEnteringPhase(1);
+        Preconditions.checkState(readOnlyRebindCount.get()==Integer.MIN_VALUE, "Partial rebind in read-only mode not supported");
+        Preconditions.checkNotNull(objectsToRebindInitial, "Objects to rebind must be set");
+
+        LOG.debug("Partial rebind Rebinding ("+mode+") from "+rebindManager.getPersister().getBackingStoreDescription()+"...");
+
+        Builder mementoRawBuilder = BrooklynMementoRawData.builder();
+
+        /*
+         * Unmanagement is done as part of the "manage" call, entity by entity.
+         */
+
+        objectsToRebindFinal = MutableSet.of();
+        while (objectsToRebindInitial.hasNext()) {
+            BrooklynObject bo = objectsToRebindInitial.next();
+            objectsToRebindFinal.add(bo);
+            
+            if (bo instanceof Entity) {
+                // if it's an entity, add all adjuncts. (if doing some sort of pause, that's maybe not necessary...)
+                objectsToRebindFinal.addAll( ((EntityInternal)bo).getPolicies() );
+                objectsToRebindFinal.addAll( ((EntityInternal)bo).getEnrichers() );
+                objectsToRebindFinal.addAll( ((EntityInternal)bo).feeds().getFeeds() );
+            }
+        }
+        
+        // get serialization
+        for (BrooklynObject bo: objectsToRebindFinal) {
+            Memento m = ((BrooklynObjectInternal)bo).getRebindSupport().getMemento();
+            BrooklynMementoPersister p = rebindManager.getPersister();
+            String mr = ((BrooklynMementoPersisterToObjectStore)p).getMementoSerializer().toString(m);
```

Comment:
Shame we have to cast this. It makes it feel pointless to leave the old implementations of `BrooklynMementoPersister` deprecated if one will potentially get runtime exceptions using it (better to have deleted the classes to give compile-time errors, depending under what circumstances this code-path will be hit).

The `BrooklynMementoPersisterInMemory` feels like a useful thing to keep in the test packages, but maybe not even worth that if we just accept the performance overhead in our test cases of always using `BrooklynMementoPersisterToObjectStore`.